### PR TITLE
Major Performance Improvements

### PR DESCRIPTION
**Pull request by LPGhatguy**
This PR relies on stuff currently unreleased in rbx-dom libraries. All of these fixes combined should result in a 3-4x speedup for large Rojo projects.

### CODE REVIEW

File: src/snapshot/patch_apply.rs [hunk 1]
```diff
@@ -117,24 +117,24 @@ fn apply_add_child(
...
-        .metadata(snapshot.metadata)
-        .snapshot_id(snapshot.snapshot_id);
-
-    let id = tree.insert_instance(parent_id, remaining_snapshot);
+    let children = take(&mut snapshot.children);
+
+    // If an object we're adding has a non-null referent, we'll note this
+    // instance down as needing to be revisited later.
+    let has_refs = snapshot.properties.values().any(|value| match value {
+        Variant::Ref(value) => value.is_some(),
```
Comment (Kampfkarren) [inline src/snapshot/patch_apply.rs:128]: `matches!(value, Variant::Ref(Some(_))` Clippy lints this, but it might not check is_some() checks

Comment (LPGhatguy) [reply src/snapshot/patch_apply.rs:128]: I think this is a clippy false positive. `value` is not an `Option`, it's actually just a `Ref`, which has `is_some` and `is_none` methods.

Comment (Kampfkarren) [reply src/snapshot/patch_apply.rs:128]: Oh, I meant that clippy *should* catch this, under my assumption it was an Option. My mistake.

Comment (LPGhatguy) [reply src/snapshot/patch_apply.rs:128]: That's probably an indication that these method names are bad, but I didn't know what else to call them, lol!

**Pull request merged**